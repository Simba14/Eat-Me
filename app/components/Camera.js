import React, { Component } from 'react';
import { View, StyleSheet, TouchableHighlight } from 'react-native';
import Camera from 'react-native-camera';

class Camera extends Component {

  clickedMe() {
    alert("was touched");
  }

  render() {
    return (
      <View>
        <Camera
          ref={(cam) => {
            this.camera = cam;
          }}
          style={styles.preview}
          aspect={Camera.constants.Aspect.fill}>
          <TouchableHighlight onPress={this.clickedMe.bind(this)}>
            <View style={styles.cameraButton}></View>
          </TouchableHighlight>
        </Camera>
      </View>
    );
  };

}

const styles = StyleSheet.create({
  preview: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center'
  },
  cameraButton: {
    height: 50,
    width: 50,
    backgroundColor: 'pink'
  }
})

export default Camera;
