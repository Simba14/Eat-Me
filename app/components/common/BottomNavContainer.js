import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';

class BottomNavContainer extends Component {
  render() {
    return (
      <View style={[styles.bottomNavContainer, this.props.style]}>
        {this.props.children}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  bottomNavContainer: {
    flex: 1.5,
    flexDirection: 'row',
    justifyContent: 'center',
  }
});

export { BottomNavContainer };
