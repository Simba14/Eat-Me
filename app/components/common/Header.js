import React from 'react';
import { View, Text } from 'react-native';

const Header = (props) => {
  const { viewStyle } = styles;
  return (
    <View style={viewStyle}>
      <Text>{props.headerText}</Text>
    </View>
  );
};

const styles = {
  viewStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 70,
    paddingTop: 15,
    borderBottom: 1,
    borderBottomWidth: 0.5
  }
}

export { Header };
