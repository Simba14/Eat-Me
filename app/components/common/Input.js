import React from 'react';
import { TextInput, StyleSheet } from 'react-native';

const Input = ({value, onChangeText, placeholder}) => {
  return (
    <TextInput
      value={value}
      onChangeText={onChangeText}
      style={ styles.textInput }
      placeholder={placeholder}
    />
  )
};

const styles = StyleSheet.create({
  textInput: {
    alignSelf: 'stretch',
    justifyContent: 'center',
    borderRadius: 5,
    borderWidth: 1,
    borderColor: 'grey',
    backgroundColor: 'white',
    marginLeft: 20,
    marginRight: 20,
    marginBottom: 10,
    marginTop: 5,
    padding: 5,
    paddingLeft: 10,
    height: 40,
    fontSize: 16
  },
  labelStyle
});

export { Input };
