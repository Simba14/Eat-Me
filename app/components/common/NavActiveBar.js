import React from 'react';
import { View, StyleSheet } from 'react-native';

const NavActiveBar = () => {
  return (
    <View style={styles.viewStyle}>
    </View>
  )
}

const styles = StyleSheet.create({
  viewStyle: {
    height: 5,
    flex: 1,
    flexDirection: 'row',
    backgroundColor: '#0FB980',
    justifyContent: 'center'
    // justifyContent: 'flex-end'
  }
});

export { NavActiveBar };