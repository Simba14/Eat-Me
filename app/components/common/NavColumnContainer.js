import React, { Component } from 'react';
import { View, StyleSheet } from 'react-native';

class NavColumnContainer extends Component {
  render() {
    return (
      <View style={[styles.navColumnContainer, this.props.style]}>
        {this.props.children}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  navColumnContainer: {
    // flexDirection: 'column',
    // flexWrap: 'wrap',
    backgroundColor: 'blue',
  }
});

export { NavColumnContainer };
