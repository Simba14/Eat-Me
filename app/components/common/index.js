export * from './Header';
export * from './Input';
export * from './LowerContainer';
export * from './SubmitButton';
export * from './UpperContainer';
export * from './ViewContainer';
export * from './BottomNavContainer';
export * from './NavItem';
