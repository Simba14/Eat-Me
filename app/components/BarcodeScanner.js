import React, { Component } from 'react';
import { View, StyleSheet, TouchableHighlight, Text, Alert } from 'react-native';
import Camera from 'react-native-camera';
import axios from 'axios';
import _ from 'lodash';

class BarcodeScanner extends Component {
  constructor() {
    super();
    this.state = {
      showCamera: true,
      scannedItem: ""
    }
  }

  render() {
      if (this.state.showCamera) {
        return (
          <View style={styles.container}>
            <Camera
              ref={(cam) => {
                this.camera = cam;
              }}
              style={styles.preview}
              aspect={Camera.constants.Aspect.fill}
              onBarCodeRead={(data) => this.requestItemInformation(data)}
            >
            </Camera>
          </View>
        );
      } else {
        return (
          <View>
            <Text>{this.state.scannedItem}</Text>
          </View>
        );
      }
  }

  requestItemInformation(data) {
    this.setState({showCamera: false});
    var barcodeNumber = data.data;
    axios.get('https://api.upcdatabase.org/json/128009a43963c119609bd223c9f249cf/' + barcodeNumber)
      .then(response => this.setState({ scannedItem: response.data.description }));
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
  },
  preview: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-start',
    marginTop: -30,
    marginBottom: 300,
  },
  capture: {
    flex: 0,
    backgroundColor: '#fff',
    borderRadius: 5,
    color: '#000',
    padding: 10,
    margin: 40
  }
});

export default BarcodeScanner;
