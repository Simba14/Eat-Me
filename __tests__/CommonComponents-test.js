import 'react-native';
import React from 'react';
import { Input, LowerContainer } from '../app/components/common';

import renderer from 'react-test-renderer';

test('renders the Input component correctly', () => {
  const tree = renderer.create(
    <Input />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders the LowerContainer component correctly', () => {
  const tree = renderer.create(
    <LowerContainer />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
