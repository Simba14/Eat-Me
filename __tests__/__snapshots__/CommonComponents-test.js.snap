// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Input component correctly 1`] = `
<TextInput
  onChangeText={undefined}
  placeholder={undefined}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "white",
      "borderColor": "grey",
      "borderRadius": 5,
      "borderWidth": 1,
      "fontSize": 16,
      "height": 40,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 20,
      "marginRight": 20,
      "marginTop": 5,
      "padding": 5,
      "paddingLeft": 10,
    }
  }
  value={undefined}
/>
`;

exports[`renders the LowerContainer component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginBottom": 30,
      },
      undefined,
    ]
  }
/>
`;
